sdk/core/Azure.Core/src/Pipeline/Internal/RedirectPolicy.cs

```diff
@@ -8,17 +8,31 @@
 
 namespace Azure.Core.Pipeline
 {
-    internal sealed class RedirectPolicy : HttpPipelinePolicy
+    /// <summary>
+    /// A pipeline policy that detects a redirect response code and resends the request to the
+    /// location specified by the response.
+    /// </summary>
+    public sealed class RedirectPolicy : HttpPipelinePolicy
     {
         private readonly int _maxAutomaticRedirections;
 
-        public static RedirectPolicy Shared { get; } = new RedirectPolicy();
+        internal static RedirectPolicy Shared { get; } = new RedirectPolicy();
 
         private RedirectPolicy()
         {
             _maxAutomaticRedirections = 50;
         }
 
+        /// <summary>
+        /// Sets a value that indicates whether redirects will be automatically followed for this message.
+        /// </summary>
+        /// <param name="message"></param>
+        /// <param name="allowAutoRedirect"></param>
+        public static void AllowAutoRedirect(HttpMessage message, bool allowAutoRedirect)
```

Comment:
Should this be called something like `SetAllowAutoRedirect` or `SetAutoRedirect`?

Reply:
I considered that ... I was worried that the former was wordy, and the latter didn't mirror the .NET API naming, but you're right, it should have a Set prefix.  I'll update it, and thanks!